providers/flagd/src/main/java/dev/openfeature/contrib/providers/flagd/resolver/common/ChannelConnector.java

```diff
@@ -120,16 +106,20 @@ public GrpcConnector(
      */
     public void initialize() throws Exception {
         log.info("Initializing GRPC connection...");
-        ChannelMonitor.monitorChannelState(ConnectivityState.READY, channel, this::onReady, this::onConnectionLost);
+        monitorChannelState(ConnectivityState.READY, channel, this::onReady, this::onConnectionLost);
     }
 
     /**
      * Returns the blocking service stub for making blocking GRPC calls.
      *
      * @return the blocking service stub
      */
-    public K getResolver() {
-        return blockingStubFunction.apply(channel).withWaitForReady();
+    public K getBlockingStub() {
+        K stub = blockingStubFunction.apply(channel).withWaitForReady();
+        if (this.deadline > 0) {
+            stub = stub.withDeadlineAfter(this.deadline, TimeUnit.MILLISECONDS);
```

Comment:
I also thought about unfying this here, in my fix for the timeouts for RPC - but somehow this messed with the inprocess resolver connection, and the e2e tests have not been successful - still no clue why, but if we encounter this again, i suggest to maybe tackle this unification separately